Closes #14090 & #14074 (#14163)
metabase/metabase: docs/users-guide/05-visualizing-results.md

![Pivot table](images/visualizations/pivot.png)

This auto-pivoting is distinct from the pivot table visualization, which we cover next.

### Pivot tables

> Pivot tables allow you swap rows and columns, group data, and include aggregate data in your table. You can group one or more metrics by one or more dimensions. 

Reviewer: I think we should go with the more expected "subtotals" throughout rather than "aggregate date" even though it's true that there are exceptions like averages where it's not _really_ a total.

```suggestion
Pivot tables allow you swap rows and columns, group data, and include subtotals in your table. You can group one or more metrics by one or more dimensions. 
```

For a given query result, you can assign fields to one of three "buckets":

- Fields to use for the table rows
- Fields to use for the table columns
- Fields to use for the table values 

Let's say we ask the following question in the notebook editor:

![Pivot table notebook](images/visualizations/pivot-table-notebook.png)

> From the `Orders` table, we've summarized the count of orders and the average order total, and grouped by `User->State`, `Product->Category`, and `Created At` binned by year. Here's our question visualized as a pivot table:

Reviewer: ```suggestion
From the `Orders` table, we've summarized by the count of orders and the average order total, and grouped by `User → State`, `Product → Category`, and `Created At` binned by year. Here's our question visualized as a pivot table:
```

![Pivot table options](images/visualizations/pivot-table-options.png)

> We've assigned the dimensions `User->State` and `Product-Category` to table rows, and assigned the dimension `Created At` as our columns: the years 2016, 2017, and so on. We can drag and drop dimensions between the row and column buckets, and add aggregations to the table values bucket. For example, if we assign a field to the columns bucket, Metabase will pivot that field and render each unique value of that field as a column heading.

Reviewer: We're kind of going back and forth between "field" and "dimension" in this PR.

```suggestion
We've assigned the fields `User → State` and `Product → Category` to table rows, and assigned the `Created At` field to generate our columns: the years 2016, 2017, and so on. We can drag and drop dimensions between the row and column buckets, and add aggregations to the table values bucket. For example, if we assign a field to the columns bucket, Metabase will pivot that field and render each unique value of that field as a column heading.
```

> You can put multiple fields in each of these buckets, but note that the order of the fields changes how Metabase displays the table: each additional field will nest within the previous field. 

Reviewer: ```suggestion
You can put multiple fields in the "rows" and "columns" buckets, but note that the order of the fields changes how Metabase displays the table: each additional field will nest within the previous field. 
```

> Where it makes sense, Metabase will automatically include aggregations for grouped rows. For example, as in the image above, because we've grouped our rows first by `State`, then by `Product->Category`, Metabase will list each `Product->Category` (like Gizmo) for each `State`, and aggregate the metrics for that subgroup. For orders placed in Wisconsin, Metabase would sum the count of orders for each category, and find the average order total across all product categories. 

Reviewer: ```suggestion
Where it makes sense, Metabase will automatically include subtotals for grouped rows. For example, as in the image above, because we've grouped our rows first by `State`, then by `Product->Category`, Metabase will list each `Product->Category` (like Gizmo) for each `State`, and aggregate the metric(s) for that subgroup. For orders placed in Wisconsin, Metabase would sum the count of orders for each category, and find the average order total across all product categories in that state. 
```

#### Histograms

If you have a bar chart like Count of Users by Age, where the x-axis is a number, you'll get a special kind of chart called a **[histogram](https://www.metabase.com/learn/basics/visualizing-data/histograms.html)**, where each bar represents a range of values (called a "bin"). Note that Metabase will automatically bin your results any time you use a number as a grouping, even if you aren't viewing a bar chart. Questions that use latitude and longitude will also get binned automatically.

![Histogram](images/visualizations/histogram.png)

By default, Metabase will automatically choose a good way to bin your results. But you can change how many bins your result has, or turn the binning off entirely, by clicking on the area to the right of the column you're grouping by:

![Binning options](images/notebook/histogram-bins.png)

> Learn more about histograms(https://www.metabase.com/learn/basics/visualizing-data/histograms.html).

Reviewer: ```suggestion
[Learn more about histograms](https://www.metabase.com/learn/basics/visualizing-data/histograms.html).
```

Here's where you can choose to hide the **label** for your x- or y-axis. You can also customize the text for your axes labels here.

### Waterfall charts

Waterfall charts are a kind of bar chart useful for visualizing results that contain both positive and negative values. Each bar on a waterfall chart shows either an increase or decrease, with a final bar on the right of the chart that represents the total value.

![Waterfall chart](images/visualizations/waterfall-chart.png)

> In the example above, the waterfall chart displays "Profit" for each "Product": Apples, Bananas, Oranges, Peaches, and Mangos. From left to right, each bar indicates the change in total. The products with green bars indicate positive values (they made a profit). Peaches, however, lost money, indicated by a red bar, which signals a negative value. The purple bar at the end shows the total profit of all products combined. You can show values on each bar, and change the colors for increases and decreases.

Reviewer: We've changed the color of the Total bar, so we can remove the reference to its color.

```suggestion
In the example above, the waterfall chart displays "Profit" for each "Product:" apples, bananas, oranges, peaches, and mangos. From left to right, each bar indicates the change in total. The products with green bars indicate positive values (they made a profit). Peaches, however, lost money, indicated by a red bar, which signals a negative value. The bar at the end shows the total profit of all products combined. You can show values on each bar, and change the colors for increases and decreases.
```
